python/metatensor-core/tests/blocks.py

```diff
@@ -30,6 +30,31 @@ def block_components():
     )
 
 
+def test_constructor_errors():
+    values = np.full((3, 3, 2, 2), -1.0)
+    samples = Labels(["s"], np.array([[0], [2], [4]]))
+    components = [
+        Labels(["c_1"], np.array([[-1], [0], [1]])),
+        Labels(["c_2"], np.array([[-4], [1]])),
+    ]
+    properties = Labels(["p"], np.array([[5], [3]]))
+
+    # this works
+    _ = TensorBlock(values, samples, components, properties)
+
+    message = "`samples` must be metatensor Labels, not <class 'str'>"
```

Comment:
if you chose to use the `__repr__` in the error message, the test would need to be adapted
```suggestion
    message = "`samples` must be metatensor Labels, not str"
```